torchrl/trainers/helpers/models.py

```diff
@@ -178,6 +181,11 @@ def make_dqn_actor(
     out_features = env_specs["action_spec"].shape[0]
     actor_class = QValueActor
     actor_kwargs = {}
+
+    if _CATEGORICAL_ACTION_ENCODING:
+        actor_kwargs.update({"action_space": "categorical"})
+        out_features = env_specs["action_spec"].space.n
```

Comment:
perhaps we could have an arg in the function that indicates if categorical has to be used.
If default (e.g. `None`) then it falls back on `_CATEGORICAL_ACTION_ENCODING`.

Reply:
I changed it here to be inferred from action spec that is set during the env setup.